mcs_benchmark_data/pipelines/commonsense_qa/commonsense_qa_benchmark_transformer.py

```diff
@@ -23,125 +24,118 @@
-                )
+            benchmark_sample_uri = "{}:sample:{}".format(dataset_uri, sample["id"])
 
-                yield concept
+            correct_choice = None
 
-                choices_list = []
+            if sample_type != "test":
+                correct_choice = ans_mapping[sample["answerKey"]]
 
-                for item in sample["question"]["choices"]:
-                    choice = BenchmarkChoice(
-                        uri="",
-                        position=item["label"],
-                        text=item["text"],
-                    )
-                    choices_list.append(choice)
-                    yield choice
+            concept = BenchmarkConcept(
+                uri="{}:concept".format(benchmark_sample_uri),
+                concept=sample["question"]["question_concept"],
+            )
 
-                choices = BenchmarkChoices(uri="", choices=tuple(choices_list))
-                yield choices
+            choices_list = []
 
-                question = BenchmarkQuestion(
-                    uri="", text=sample["question"]["stem"], concepts=concept
+            for item in sample["question"]["choices"]:
+                choice = BenchmarkChoice(
+                    uri="{}:choice:{}: ".format(benchmark_sample_uri, item["label"]),
+                    position=item["label"],
+                    text=item["text"],
                 )
-                yield question
-
-                antecedent = BenchmarkAntecedent(uri="", elements=question)
-                yield BenchmarkSample(
-                    uri=sample["id"],
-                    includedInDataset=included_in_dataset,
-                    questionType=question_type,
-                    questionCategory=question_category,
-                    antecedent=antecedent,
-                    choices=choices,
-                    correctChoice=correct_choice,
-                )
-
-   
\ No newline at end of file
+                choices_list.append(choice)
+
+            choices = BenchmarkChoices(
+                uri="{}:choices".format(benchmark_sample_uri),
+                choices=tuple(choices_list),
+            )
+
+            question = BenchmarkQuestion(
+                uri="{}:question".format(benchmark_sample_uri),
+                text=sample["question"]["stem"],
+                concepts=concept,
+            )
+
+            antecedent = BenchmarkAntecedent(
+                uri="{}:antecedent".format(benchmark_sample_uri), elements=question
+            )
+            yield BenchmarkSample(
+                uri=sample["id"],
+                datasetURI=dataset_uri,
+                questionType=question_type,
+                questionCategory=question_category,
+                antecedent=antecedent,
+                choices=choices,
+                correctChoice=correct_choice,
+            )
+
+            yield question
+            yield choices
+
+            for item in choices_list:
```

Comment:
This is odd. In any case, you appear to be yielding duplicate models. You yield antecedent @ line 140, but antecedent is also part of BenchmarkSample. Either (1) antecedent should only be a nested model of BenchmarkSample, in which case you don't need to yield it separately or (2) it should be a separate model and reference BenchmarkSample by id. Whatever fits with the ontology. My usual rule of thumb is that models with their own id's should reference other model's with id's, and models without their own id's should be nested in a model that does have an id.